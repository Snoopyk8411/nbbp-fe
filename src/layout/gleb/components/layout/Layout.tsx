import { ReactNode } from 'react';
import { useRouter } from 'next/router';
import type { NextPage } from 'next';

import { MAIN_ROUTE, NOTES_ROUTE, GO_TO_PICTURE, GO_TO_NOTES } from './constants';

import styles from './layout.module.css';

const Layout = ({ children }: { children: ReactNode | NextPage }) => {
  const router = useRouter();

  const isNotesRoute: boolean = router.pathname === NOTES_ROUTE;

  const handleClickNavigation = () => {
    isNotesRoute ? router.push(MAIN_ROUTE) : router.push(NOTES_ROUTE);
  };

  return (
    <div>
      <div className={styles.header} data-test='headerComponent'>
        <button className={styles.header_title} data-testid='NavigationButton' onClick={handleClickNavigation}>
          {isNotesRoute ? GO_TO_PICTURE : GO_TO_NOTES}
        </button>
      </div>
      {children}
    </div>
  );
};

export default Layout;
