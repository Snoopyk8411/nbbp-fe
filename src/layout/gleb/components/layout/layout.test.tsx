import { ReactNode } from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Layout from 'layout/gleb/components/layout/Layout';

let mockStore: any;
const mockStoreConf = configureStore();
const useRouter = jest.spyOn(require('next/router'), 'useRouter');
const mockData = {};

describe('Note Component', () => {
  let props: { children: string };
  beforeEach(() => {
    mockStore = mockStoreConf(mockData);
    props = {
      children: 'Component',
    };
  });

  it('Note snapshot', () => {
    useRouter.mockImplementationOnce(() => ({
      pathname: '/contributors/gleb/notes',
    }));

    const component = renderer.create(
      <Provider store={mockStore}>
        <Layout {...props} />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
