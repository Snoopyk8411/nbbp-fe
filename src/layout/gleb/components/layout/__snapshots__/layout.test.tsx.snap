// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Component Note snapshot 1`] = `
<div>
  <div
    className="header"
    data-test="headerComponent"
  >
    <button
      className="header_title"
      data-testid="HeaderTitle"
      onClick={[Function]}
    >
      Go to the Picture
    </button>
  </div>
  Component
</div>
`;
